content/doc/book/installing/_docker.adoc

```diff
@@ -13,264 +13,129 @@ _run-jenkins-in-docker.adoc).
-<8> Maps the `/certs/client` directory inside the container to
-a Docker volume named `jenkins-docker-certs` as created above.
-<9> Maps the `/var/jenkins_home` directory inside the container to the Docker
-volume named `jenkins-data` as created above. This will allow for other Docker
-containers controlled by this Docker container's Docker daemon to mount data
-from Jenkins.
-<10> ( _Optional_ ) Exposes the Docker daemon port on the host machine. This is
-useful for executing `docker` commands on the host machine to control this
-inner Docker daemon.
-<11> The `docker:dind` image itself. This image can be downloaded before running
-by using the command: `docker image pull docker:dind`.
-+
-*Note:* If copying and pasting the command snippet above does not work, try
-copying and pasting this annotation-free version here:
+FROM jenkins/jenkins:2.249.1-lts
+USER root
+RUN apt-get update && apt-get install -y \
+       apt-transport-https \
+       ca-certificates \
+       curl \
+       gnupg2 \
+       software-properties-common
+RUN curl -fsSL https://download.docker.com/linux/debian/gpg | apt-key add -
+RUN apt-key fingerprint 0EBFCD88
+RUN add-apt-repository \
+       "deb [arch=amd64] https://download.docker.com/linux/debian \
+       $(lsb_release -cs) \
+       stable"
+RUN apt-get update && apt-get install -y docker-ce-cli
+USER jenkins
+RUN /usr/local/bin/install-plugins.sh blueocean:1.24.0
+
+ENTRYPOINT ["/sbin/tini", "--", "/usr/local/bin/jenkins.sh"]
+----
+.. Build a new docker image from this Dockerfile and assign the image meaningful name, e.g. "myjenkins:1.0":
 +
-[source,bash]
+[source]
 ----
-docker container run --name jenkins-docker --rm --detach \
-  --privileged --network jenkins --network-alias docker \
-  --env DOCKER_TLS_CERTDIR=/certs \
-  --volume jenkins-docker-certs:/certs/client \
-  --volume jenkins-data:/var/jenkins_home \
-  --publish 2376:2376 docker:dind
+docker build -t "myjenkins:1.0" .
 ----
-. Download the `jenkinsci/blueocean` image and run it as a container in Docker
-  using the following
-  link:https://docs.docker.com/engine/reference/commandline/container_run/[`docker container run`]
+Keep in mind that described above process will automatically download official Jenkins Docker image 
+if this hasn't been done before.
+
+. Run your own `myjenkins:1.0` image as a container in Docker using the
+  following
+  link:https://docs.docker.com/engine/reference/run/[`docker run`]
   command:
 +
-[source,bash]
+[source]
 ----
-docker container run \
-  --name jenkins-blueocean \# <1>
-  --rm \# <2>
-  --detach \# <3>
-  --network jenkins \# <4>
-  --env DOCKER_HOST=tcp://docker:2376 \# <5>
-  --env DOCKER_CERT_PATH=/certs/client \
-  --env DOCKER_TLS_VERIFY=1 \
-  --publish 8080:8080 \# <6>
-  --publish 50000:50000 \# <7>
-  --volume jenkins-data:/var/jenkins_home \# <8>
-  --volume jenkins-docker-certs:/certs/client:ro \# <9>
-  jenkinsci/blueocean# <10>
+docker run --rm -u root --name jenkins-dkr-blueocean \
```

Comment:
Jenkins isn't designed to be run as the root user as far as I know.  Running as root provides too many permissions.  This runs the  controller as root.  I don't think we can require that of a user, even in a test environment.

I assume you're running as root so that the Jenkins process can access the docker command line.  I believe the Docker image definition can be modified to not run as the root user, though my initial attempts did not find the correct sequence.
```suggestion
docker run --rm --name jenkins-dkr-blueocean \
```

Reply:
@MarkEWaite , exactly right, this approach enables access to docker cli inside the docker jenkins image.
There were many discussions about alternative sequence, like [this one](https://forums.docker.com/t/docker-not-found-in-jenkins-pipeline/31683/27) and others, and it looks running as root is the most recommended solution for our requirements.

FTR CloudBees recommends very similar approach as well for running Jenkins distribution in docker:
![image](https://user-images.githubusercontent.com/2159948/96287374-b02a3800-0f96-11eb-944b-767206ba9aee.png)

Reply:
That looks to be outdated information. I updated the initial docs on jenkins.io about this since CloudBees was re-using some of the instructions, but it looks like this part wasn't updated downstream.

The important security consideration here is that Jenkins agents that run `docker` commands should not be using the same Docker daemon as the one that the Jenkins controller is running in. Mounting the outer Docker socket like that shares the same Docker daemon which allows agents and controllers equal permissions (typically not what you want). On the other hand, if you're running on agent on a different machine or VM than the controller, then mounting the docker socket there doesn't compromise security. When running everything all in one VM, though, it's important to run the two separate layers of Docker since agents will typically desire privileged access to build and run containers.

Reply:
Thanks for the comments.  Our goal with the changes from @vsilverman is to replace jenkinsci/blueocean with a Docker image that the user generates based on the official Jenkins image.  I think we can use the dind image technique that you used with a variant of the custom image definition that Vlad has created.

The tutorial step previously said:

> docker container run --name jenkins-tutorial --rm --detach \
>   --network jenkins --env DOCKER_HOST=tcp://docker:2376 \
>   --env DOCKER_CERT_PATH=/certs/client --env DOCKER_TLS_VERIFY=1 \
>   --volume jenkins-data:/var/jenkins_home \
>   --volume jenkins-docker-certs:/certs/client:ro \
>   --volume "$HOME":/home --publish 8080:8080 jenkinsci/blueocean

I think we may be able to get the same results with Vlad's derivative of the official Jenkins image with:

> docker container run --name jenkins-tutorial --rm --detach \
>   --network jenkins --env DOCKER_HOST=tcp://docker:2376 \
>   --env DOCKER_CERT_PATH=/certs/client --env DOCKER_TLS_VERIFY=1 \
>   --volume jenkins-data:/var/jenkins_home \
>   --volume jenkins-docker-certs:/certs/client:ro \
>   --volume "$HOME":/home --publish 8080:8080 myjenkins:1.0

It worked for me when I ran the java-app-with-maven example on Linux.

@jvz do you see any clear gaps in my idea of replacing the jenkinsci/blueocean image with a Docker image the user creates that contains blueocean?